exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -289,23 +292,58 @@ private void checkTransaction(TransactionMessage transactionMessage) {
 
   /**
    * Performs a given function with a snapshot of the current database state (i.e., the one that
-   * corresponds to the latest committed block). In-pool (not yet processed) transactions are also
-   * accessible with it in {@linkplain Blockchain#getTxMessages() blockchain}.
+   * corresponds to the latest committed block) and returns a result of its execution. In-pool
+   * (not yet processed) transactions are also accessible with it in
+   * {@linkplain Blockchain#getTxMessages() blockchain}.
    *
    * @param snapshotFunction a function to execute
    * @param <ResultT> a type the function returns
    * @return the result of applying the given function to the database state
    */
-  public <ResultT> ResultT withSnapshot(Function<Snapshot, ResultT> snapshotFunction) {
-    try (Cleaner cleaner = new Cleaner("TestKit#withSnapshot")) {
-      long snapshotHandle = nativeCreateSnapshot(nativeHandle.get());
-      Snapshot snapshot = Snapshot.newInstance(snapshotHandle, cleaner);
+  public <ResultT> ResultT applySnapshot(Function<Snapshot, ResultT> snapshotFunction) {
```

Comment:
The docs must mention why these methods exist on top of #getSnapshot and explain when they shall be used instead of #getSnapshot.